Install on MacOS/Python 3.11.5 fails

**Describe the bug**
Install on MacOS (Ventura 13.5.2) for Python 3.11.5 currently fails because two dependencies fail to install.

**To Reproduce**
Steps to reproduce the behavior:
1. `conda create -n ark python=3.11`
2. `conda install cython`
3. `conda install .`

**Expected behavior**
```
pip install .
 ```
should succeed inside a new conda environment for Python 3.11.

**Error Message**
1. For tables: same stack trace as in https://github.com/PyTables/PyTables/issues/1040
2. For pyarrow:
```Building wheel for pyarrow (pyproject.toml) ... error
  error: subprocess-exited-with-error
  ...
  -- Running cmake for pyarrow
  ...
  -- Checking for module 'arrow'
     ...
      --   No package 'arrow' found
      CMake Error at /opt/homebrew/Cellar/cmake/3.27.6/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
        Could NOT find Arrow (missing: ARROW_INCLUDE_DIR ARROW_LIB_DIR
        ARROW_FULL_SO_VERSION ARROW_SO_VERSION)
      Call Stack (most recent call first):
        /opt/homebrew/Cellar/cmake/3.27.6/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
        cmake_modules/FindArrow.cmake:450 (find_package_handle_standard_args)
        cmake_modules/FindArrowPython.cmake:46 (find_package)
        CMakeLists.txt:236 (find_package)
      
      -- Configuring incomplete, errors occurred!
      error: command '/opt/homebrew/bin/cmake' failed with exit code 1
      [end of output]
  ```

**Is this a Blocking Issue**
Work-around for tables install, see: https://github.com/PyTables/PyTables/issues/1040
```pip install git+https://github.com/PyTables/PyTables```

Work-around for pyarrow:
First run
```conda install pyarrow``` (installs pyarrow 11.0)
Then modify setup.py (no longer pinning to pyarrow==9.0.0)

I don't know if it's OK to unpin pyarrow, but this does make the `pip install` succeed.

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -63,7 +63,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        python-version: ['3.8', '3.9', '3.10']
+        python-version: ['3.8', '3.9', '3.10', '3.x']
     container:
       image: chapel/chapel:1.31.0
     steps:

--- a/Makefile
+++ b/Makefile
@@ -125,7 +125,7 @@ install-hdf5:
 	rm -rf $(HDF5_BUILD_DIR)
 	echo '$$(eval $$(call add-path,$(HDF5_INSTALL_DIR)))' >> Makefile.paths
 
-ARROW_VER := 9.0.0
+ARROW_VER := 11.0.0
 ARROW_NAME_VER := apache-arrow-$(ARROW_VER)
 ARROW_FULL_NAME_VER := arrow-apache-arrow-$(ARROW_VER)
 ARROW_BUILD_DIR := $(DEP_BUILD_DIR)/$(ARROW_FULL_NAME_VER)

--- a/arkouda-env-dev.yml
+++ b/arkouda-env-dev.yml
@@ -16,7 +16,7 @@ dependencies:
   - pip
   - types-tabulate
   - pytables>=3.7.0,<3.9.0
-  - pyarrow==9.0.0
+  - pyarrow==11.0.0
   - libiconv
   - libidn2
 

--- a/arkouda-env.yml
+++ b/arkouda-env.yml
@@ -16,7 +16,7 @@ dependencies:
   - pip
   - types-tabulate
   - pytables>=3.7.0,<3.9.0
-  - pyarrow==9.0.0
+  - pyarrow==11.0.0
   - libiconv
   - libidn2
 

--- a/pydoc/requirements.txt
+++ b/pydoc/requirements.txt
@@ -13,7 +13,7 @@ hdf5==1.12.2
 pip
 types-tabulate
 tables>=3.7.0,<3.9.0
-pyarrow==9.0.0
+pyarrow==11.0.0
 libiconv
 libidn2
 

--- a/pydoc/setup/REQUIREMENTS.md
+++ b/pydoc/setup/REQUIREMENTS.md
@@ -31,7 +31,7 @@ The following python packages are required by the Arkouda client package.
 - `pip`
 - `types-tabulate`
 - `tables>=3.7.0,<3.9.0`
-- `pyarrow>=1.0.1`
+- `pyarrow==11.0.0`
 
 ### Developer Specific
 

--- a/setup.py
+++ b/setup.py
@@ -143,7 +143,7 @@ setup(
         'pip',
         'types-tabulate',
         'tables>=3.7.0,<3.9.0',
-        'pyarrow==9.0.0',
+        'pyarrow==11.0.0',
     ],
 
     # List additional groups of dependencies here (e.g. development

